wfe2/wfe.go

```diff
@@ -2401,7 +2401,7 @@ func (wfe *WebFrontEndImpl) NewOrder(
 		return
 	}
 
-	refundLimits, err := wfe.checkNewOrderLimits(ctx, acct.ID, names, isRenewal)
+	refundLimits, err := wfe.checkNewOrderLimits(ctx, acct.ID, names, isRenewal || isARIRenewal)
```

Comment:
Good catch! If I understand this bug correctly, currently we're only providing an exemption from `NewOrdersPerAccount` and `CertificatesPerDomain` for legacy renewals and some ARI renewals that also happen to be legacy renewals (renewals containing the same set of names).

Instead, we want to orders that qualify under the more stringent ARI exemption to be exempt from all rate limits, not just `NewOrdersPerAccount` and `CertificatesPerDomain`. If this is the case, I think we need to wrap this in a conditional, like so:

```go
if !ariRenewal {
    refundLimits, err := wfe.checkNewOrderLimits(ctx, acct.ID, names, isRenewal)
}
```